tutor/src/components/buttons/teacher-become-student.js

```diff
@@ -86,11 +86,11 @@ class TeacherBecomesStudent extends React.Component {
     );
   }
 
-  @action async becomeStudentInPeriod(period) {
+  async becomeStudentInPeriod(period) {
     const { course } = this.props;
     this.isCreating = true;
-    this.context.router.history.push(`/course/${course.id}`);
-    await period.becomeStudent();
+    const role = await period.findOrCreateTeacherStudentRole();
+    this.context.router.history.push(`/course/${course.id}/become/${role.id}`);
```

Comment:
Doesn't this need to call `role.become()` now? Or does pushing stuff to history actually call the renderer?

Reply:
no, we now call role.become in the new route that lives in the conditionalRoutes.  Calling it here caused the app to start re-rendering while it was still on the teacher dashboard and then it was a race if the route switch would happen or the dashboard would re-render

Reply:
OK it was just not clear to me what was actually calling the renderer.